backend/app/src/main/kotlin/io/tolgee/api/v2/controllers/PatController.kt

```diff
@@ -0,0 +1,113 @@
+package io.tolgee.api.v2.controllers
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.tags.Tag
+import io.tolgee.api.v2.hateoas.pat.PatModel
+import io.tolgee.api.v2.hateoas.pat.PatModelAssembler
+import io.tolgee.api.v2.hateoas.pat.RevealedPatModel
+import io.tolgee.api.v2.hateoas.pat.RevealedPatModelAssembler
+import io.tolgee.controllers.IController
+import io.tolgee.dtos.request.pat.CreatePatDto
+import io.tolgee.dtos.request.pat.RegeneratePatDto
+import io.tolgee.dtos.request.pat.UpdatePatDto
+import io.tolgee.exceptions.PermissionException
+import io.tolgee.model.Pat
+import io.tolgee.security.AuthenticationFacade
+import io.tolgee.security.patAuth.DenyPatAccess
+import io.tolgee.service.PatService
+import org.springdoc.api.annotations.ParameterObject
+import org.springframework.data.domain.Pageable
+import org.springframework.data.web.PagedResourcesAssembler
+import org.springframework.hateoas.PagedModel
+import org.springframework.http.HttpStatus
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import javax.validation.Valid
+
+@RestController
+@RequestMapping("/v2/pats")
+@Tag(name = "Personal Access Tokens")
+class PatController(
+  private val patService: PatService,
+  private val patModelAssembler: PatModelAssembler,
+  @Suppress("SpringJavaInjectionPointsAutowiringInspection")
+  private val pagedResourcesAssembler: PagedResourcesAssembler<Pat>,
+  private val authenticationFacade: AuthenticationFacade,
+  private val revealedPatModelAssembler: RevealedPatModelAssembler
+) : IController {
+
+  @GetMapping(value = [""])
+  @Operation(summary = "Get all Personal Access Tokens")
+  fun getAll(
+    @ParameterObject pageable: Pageable
+  ): PagedModel<PatModel> {
+    return pagedResourcesAssembler.toModel(
+      patService.findAll(authenticationFacade.userAccount.id, pageable),
+      patModelAssembler
+    )
+  }
+
+  @GetMapping(value = ["/{id:[0-9]+}"])
+  @Operation(summary = "Get Personal Access Token")
+  fun get(
+    @PathVariable id: Long
+  ): PatModel {
+    return patModelAssembler.toModel(patService.get(id))
```

Comment:
Good point!

Reply:
Again, when the user has no access to some resource, I am returning 403 rather than 404. I know 404 prevents attackers from enumerating, but what the attacker can do if they know there is PAT with some ID existing? 

It's maybe less secure but it's also less confusing. When I've got an 404 error I would think I am providing wrong id, but actually I am just using wrong account.

Reply:
Hm, considering this only exposes the ID it should be safe to 403 out, you're right.